Assets/IntegrationTests/TestGenericQuery.cs

```diff
@@ -1,25 +1,31 @@
 namespace Shopify.Unity.Tests
 {
     using UnityEngine;
+	using UnityEngine.TestTools;
+	using NUnit.Framework;
     using Shopify.Unity;
+	using System.Collections;
 
-    [IntegrationTest.DynamicTest ("TestScene")]
-    [IntegrationTest.SucceedWithAssertions]
     public class TestGenericQuery : MonoBehaviour {
-        void Start() {
+		[UnityTest]
+		public IEnumerator LoadShopName() {
             ShopifyBuy.Init("351c122017d0f2a957d32ae728ad749c", "graphql.myshopify.com");
+			bool hadResult = false;
 
             ShopifyBuy.Client().Query(
                 (q) => q.shop(s => s
                     .name()
                 ), 
                 (data, error) => {
-                    IntegrationTest.Assert(null == error, "No errors");
-                    IntegrationTest.Assert("graphql" == data.shop().name(), "Shop name was \"graphql\"");
-
-                    IntegrationTest.Pass();
+					hadResult = true;
+					Assert.IsNull(error, "No errors");
+					Assert.AreEqual("graphql", data.shop().name(), "Shop name was \"graphql\"");
                 }
             );
+
+			yield return Utils.GetWaitMaxQueryDuration ();
```

Comment:
does this work if you are running in play mode? I know in edit mode you have to return `null` here

Reply:
Will this always wait some large amount of time? What is the request ends sooner? Can this GetWaitMaxQueryDuration take the hadResult bool and stop waiting if that is true?